src/components/RadioSelect/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { slippageItem } from '../../pages/PageMarkets/PageMarketsContent';
+
+import s from './style.module.scss';
+
+interface IRadioSelect {
+  items: slippageItem[];
+  onChecked: (value: string | number, arr: slippageItem[]) => void;
+  percent?: boolean;
+  custom?: boolean;
+  customValue?: number;
+  wrapperClassName?: string;
+  className?: string;
+  customPlaceholder?: string;
+}
+
+export const RadioSelect: React.FC<IRadioSelect> = ({
+  items,
+  percent = false,
+  custom = false,
+  customValue,
+  onChecked,
+  wrapperClassName,
+  className,
+  customPlaceholder,
+}) => {
+  const [value, setValue] = React.useState<any>(customValue || '');
+
+  const handleChangeCheck = (index: number) => {
+    let newSlippage;
+    if (index === -1) {
+      newSlippage = items.map((item) => {
+        return {
+          ...item,
+          checked: false,
+        };
+      });
+      onChecked(0, newSlippage);
+    } else {
+      newSlippage = items.map((item, i) => {
+        return {
+          ...item,
+          checked: index === i,
+        };
+      });
+      onChecked(items[index].text, newSlippage);
+      setValue('');
+    }
```

Comment:
перепиши блок без использования if

```suggestion
  const newSlippage = items.map((item, i) => ({checked: index === -1 ? false : index === i, ...item}));
  onChecked(index === -1 ? 0 : items[index].text, newSlippage);
  setValue('');
```